src/main/java/core/basesyntax/lib/Injector.java

```diff
@@ -0,0 +1,40 @@
+package core.basesyntax.lib;
+
+import core.basesyntax.dao.BetDao;
+import core.basesyntax.dao.UserDao;
+import core.basesyntax.exceptions.AnnotationException;
+import core.basesyntax.factory.Factory;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+
+public class Injector {
+    public static Object getInstance(Class clazz) throws NoSuchMethodException,
+            IllegalAccessException, InvocationTargetException, InstantiationException {
+        Constructor constructor = clazz.getDeclaredConstructor();
+        Object instance = constructor.newInstance();
+
+        Field[] declaredFields = clazz.getDeclaredFields();
+        validate();
+        for (Field field : declaredFields) {
+            if (field.getAnnotation(Inject.class) != null) {
+                field.setAccessible(true);
+                if (field.getType() == BetDao.class) {
+                    field.set(instance, Factory.getBetDao());
+                } else if (field.getType() == UserDao.class) {
+                    field.set(instance, Factory.getUserDao());
+                }
+            }
+        }
+        return instance;
+    }
+
+    private static void validate() {
+        if (Factory.getBetDao().getClass().isAnnotationPresent(Dao.class)
+                && Factory.getUserDao().getClass().isAnnotationPresent(Dao.class)) {
+            return;
+        }
+        throw new AnnotationException("THIS IS STILL IN BETA, "
```

Comment:
this message is uninformative

Reply:
Is it really THAT important in the task that is just an emulation of a real behavior and wrong in the first place??

Reply:
point of all tasks is to learn how to write correct code, with this logic you can do whatever you want in your code  because task isn't real